metricflow/test/model/test_where_filter_spec.py

```diff
@@ -98,6 +99,94 @@ def test_time_dimension_in_filter(  # noqa: D
     )
 
 
+def test_date_part_in_filter(  # noqa: D
+    column_association_resolver: ColumnAssociationResolver,
+) -> None:
+    where_filter = PydanticWhereFilter(where_sql_template="{{ Dimension('metric_time').date_part('year') }} = '2020'")
+
+    where_filter_spec = WhereSpecFactory(
+        column_association_resolver=column_association_resolver,
+    ).create_from_where_filter(where_filter)
+
+    assert where_filter_spec.where_sql == "metric_time__extract_year = '2020'"
+    assert where_filter_spec.linkable_spec_set == LinkableSpecSet(
+        dimension_specs=(),
+        time_dimension_specs=(
+            TimeDimensionSpec(
+                element_name="metric_time",
+                entity_links=(),
+                time_granularity=TimeGranularity.DAY,
+                date_part=DatePart.YEAR,
+            ),
+        ),
+        entity_specs=(),
+    )
+
+
+@pytest.mark.parametrize(
+    "where_sql",
+    (
+        ("{{ TimeDimension('metric_time', 'WEEK', date_part_name='year') }} = '2020'"),
+        ("{{ Dimension('metric_time').date_part('year').grain('WEEK') }} = '2020'"),
+        ("{{ Dimension('metric_time').grain('WEEK').date_part('year') }} = '2020'"),
```

Comment:
I hope these aren't case-sensitive. I may add some additional tests on after we merge.